src/Functions/getHttpHeader.cpp

```diff
@@ -0,0 +1,92 @@
+#include <Functions/IFunction.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <DataTypes/DataTypeString.h>
+#include <Columns/ColumnString.h>
+#include <Interpreters/Context.h>
+#include <Common/CurrentThread.h>
+#include <Core/Field.h>
+#include <Poco/Net/NameValueCollection.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+    extern const int FUNCTION_NOT_ALLOWED;
+}
+
+namespace
+{
+
+/** Get the value of parameter in http headers.
+  * If there no such parameter or the method of request is not
+  * http, the function will return empty string.
+  */
+class FunctionGetHttpHeader : public IFunction
```

Comment:
```suggestion
class FunctionGetHttpHeader : public IFunction, WithContext
```
and then you save `context` from constructor and use that instead of `DB::CurrentThread::getQueryContext`

Reply:
I've tried your suggestion, but it throw a fatal error.So I think the status quo is fine.

Reply:
what kind of fatal error? it shouldn't throw one

Reply:
<img width="1764" alt="image" src="https://github.com/ClickHouse/ClickHouse/assets/87509571/5f674a2d-67a8-42e1-b78d-71a5453fb145">

Reply:
I konw what happend, it's my fault. I will modify it .